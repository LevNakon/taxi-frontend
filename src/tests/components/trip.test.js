import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import Trip from '../../components/trip';

describe('Trip --- test render correctly component', () => {
    const initialState = {
        tripState: {
            message: '',
            success: true
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = shallow(<Trip store={store} />);
    });

    it('render correctly component Trip', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('check props matches with initialState', () => {
        expect(wrapper.dive().prop('message')).toEqual(initialState.tripState.message);
        expect(wrapper.dive().prop('success')).toEqual(initialState.tripState.success);
    });
});