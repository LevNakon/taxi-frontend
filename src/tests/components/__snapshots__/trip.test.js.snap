// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trip --- test render correctly component render correctly component Trip 1`] = `
<ContextProvider
  value={null}
>
  <Trip
    message=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    success={true}
    tripCreateWatcher={[Function]}
    tripNull={[Function]}
  />
</ContextProvider>
`;
